TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/ResultsToLogsConsumer.java

```diff
@@ -33,15 +44,17 @@ public class ResultsToLogsConsumer implements BiConsumer<SourceTargetCaptureTupl
 
     private final Logger tupleLogger;
     private final Logger progressLogger;
-    private final IJsonTransformer tupleTransformer;
+
+    private final ThreadSafeTransformerWrapper threadSafeTransformer;
 
     private final AtomicInteger tupleCounter;
 
-    public ResultsToLogsConsumer(Logger tupleLogger, Logger progressLogger, IJsonTransformer tupleTransformer) {
+    public ResultsToLogsConsumer(Logger tupleLogger, Logger progressLogger, Supplier<IJsonTransformer> tupleTransformerSupplier) {
         this.tupleLogger = tupleLogger != null ? tupleLogger : LoggerFactory.getLogger(OUTPUT_TUPLE_JSON_LOGGER);
         this.progressLogger = progressLogger != null ? progressLogger : makeTransactionSummaryLogger();
         tupleCounter = new AtomicInteger();
-        this.tupleTransformer = tupleTransformer != null ? tupleTransformer : NOOP_JSON_TRANSFORMER ;
+        Supplier<IJsonTransformer> jsonTransformerSupplier = tupleTransformerSupplier != null ? tupleTransformerSupplier : () -> NOOP_JSON_TRANSFORMER ;
```

Comment:
same comment as above the NOOP transformer is already threadsafe, so you should be able to invert this